router/Router.js

```diff
@@ -182,22 +183,65 @@ class Router {
                 return cb(errors.NotImplemented);
             }
             utapiRequest.setRoute(route);
-            return this._processSecurityChecks(utapiRequest, route,
-                requestData, cb);
+            return this._validateRoute(utapiRequest, route, requestData, cb);
         });
     }
 
     /**
-     * Process security checks according to the route
+     * Send authentication and authorization request to vault
+     * @param {UtapiRequest} utapiRequest - Utapi request object
+     * @param {function} cb - Callback (err)
+     * @return {undefined}
+     */
+    _authSquared(utapiRequest, cb) {
+        const log = utapiRequest.getLog();
+        const authHeader = utapiRequest.getRequestHeaders().authorization;
+        if (!authHeader || !authHeader.startsWith('AWS4')) {
+            log.trace('missing auth header for v4 auth');
+            return cb(errors.InvalidRequest.customizeDescription('Must ' +
+                'use Auth V4 for this request.'));
+        }
+        // resourceType will either be "buckets", "accounts" or "users"
+        const resourceType = utapiRequest.getResource();
+        const validator = utapiRequest.getValidator();
+        // specific resources will be names of buckets, accounts or users
+        const specificResources = validator.get(resourceType);
+        // TODO: once vault accepts an array of request contexts, modify this
+        // to remove condition and to send array of requestContexts
+        if (specificResources.length > 1) {
+            return cb(errors.NotImplemented.customizeDescription('Users ' +
```

Comment:
this message look strange, i understand users can't have multiples specific resources (agreed). But what about accounts ? Actually, this error will be raised for both user or account right ? Can be confusing :)